src/utils/scandir.c

```diff
@@ -57,8 +46,43 @@ int scan_directory(list *dir_list, const char *dir) {
 #endif
 }
 
+bool scan_directory_for_file(char *path, size_t path_size) {
+#if defined(_WIN32) || defined(WIN32)
+    return true;
+#else
+    char *path_dup = omf_strdup(path);
+    char *path_dup2 = omf_strdup(path);
+    char *fn = basename(path_dup);
+    char *directory = dirname(path_dup2);
+
+    list dir_list;
+    list_create(&dir_list);
+    if(scan_directory(&dir_list, directory) != 0) {
+        list_free(&dir_list);
+        omf_free(path_dup);
+        omf_free(path_dup2);
+        return false;
+    }
+    iterator it;
+    list_iter_begin(&dir_list, &it);
+    const char *iter_fn;
+    size_t fn_len = strlen(fn);
+    bool found = false;
+    foreach(it, iter_fn) {
+        if(strlen(iter_fn) == fn_len && omf_strncasecmp(fn, iter_fn, fn_len) == 0) {
+            snprintf(path, path_size, "%s/%s", directory, iter_fn);
+            found = true;
+            break;
+        }
+    }
+    omf_free(path_dup);
+    omf_free(path_dup2);
+    list_free(&dir_list);
+    return found;
+#endif
+}
+
 int scan_directory_prefix(list *dir_list, const char *dir, const char *prefix) {
-    check_valid_directory(dir);
```

Comment:
Why was this removed?

Reply:
It is asserting that a directory ends in the path delimiter, but that is a wrong assertion, and no reason to crash the game. It was crashing the game for me on some path that I don't remember anymore.

Reply:
So the point of this change is that sanitizing paths should not be an assert crash, it can be still fixed at this point.